### PR TITLE
Add build validation scripts version as custom value

### LINKED ISSUE #230: Add scripts version as a custom value on the Build Scan
We should add the version of the build validation scripts used to execute the experiments as a custom value on the Build Scan (e.g. `Build validation scripts version=2.1`)

### PR DESCRIPTION
**Pull request by erichaagdev**
This PR adds the build validation scripts version as a custom value to executed builds. 

- Gradle: https://ge.solutions-team.gradle.com/s/wzntvwjjdcajm/custom-values#L0
- Maven: https://ge.solutions-team.gradle.com/s/6eymkkmgysalo/custom-values#L0

fixes #230

### CODE REVIEW

**Review by jthurne — APPROVED** [review]
Nice!

File: components/capture-build-scan-url-maven-extension/src/main/java/com/gradle/CaptureBuildScansListener.java [hunk 1]
```diff
@@ -46,6 +46,9 @@ private static void addCustomDataOnBuildFinished(BuildScanApi buildScan) {
 
             String runId = System.getProperty("com.gradle.enterprise.build-validation.runId");
             addCustomValueAndSearchLink(buildScan, "Experiment run id", runId);
+
+            String version = System.getProperty("com.gradle.enterprise.build-validation.version");
+            buildScan.value("Build validation scripts version", version);
```
Comment (etiennestuder) [inline components/capture-build-scan-url-maven-extension/src/main/java/com/gradle/CaptureBuildScansListener.java:51]: I would remove the `version` part.

File: components/scripts/gradle/gradle-init-scripts/configure-gradle-enterprise.gradle [hunk 2]
```diff
@@ -112,6 +113,8 @@ def addBuildScanCustomData = { def buildScan ->
     buildScan.tag(expId)
 
     addCustomValueAndSearchLink(buildScan, "Experiment run id", runId)
+
+    buildScan.value("Build validation scripts version", version)
```
Comment (etiennestuder) [inline components/scripts/gradle/gradle-init-scripts/configure-gradle-enterprise.gradle:117]: Same comment as above.

File: components/scripts/lib/gradle.sh [hunk 3]
```diff
@@ -34,6 +34,7 @@ invoke_gradle() {
     -Dcom.gradle.enterprise.build-validation.expId="${EXP_SCAN_TAG}"
     -Dcom.gradle.enterprise.build-validation.runId="${RUN_ID}"
     -Dcom.gradle.enterprise.build-validation.runNum="${run_num}"
+    -Dcom.gradle.enterprise.build-validation.version="${SCRIPT_VERSION}"
```
Comment (etiennestuder) [inline components/scripts/lib/gradle.sh:37]: I would rename the property to `com.gradle.enterprise.build-validation.scriptsVersion`.

File: components/scripts/lib/maven.sh [hunk 4]
```diff
@@ -58,6 +58,7 @@ invoke_maven() {
     -Dcom.gradle.enterprise.build-validation.expId="${EXP_SCAN_TAG}"
     -Dcom.gradle.enterprise.build-validation.runId="${RUN_ID}"
     -Dcom.gradle.enterprise.build-validation.runNum="${run_num}"
+    -Dcom.gradle.enterprise.build-validation.version="${SCRIPT_VERSION}"
```
Comment (etiennestuder) [inline components/scripts/lib/maven.sh:61]: I would rename the property to `com.gradle.enterprise.build-validation.scriptsVersion`.

File: components/scripts/gradle/gradle-init-scripts/configure-gradle-enterprise.gradle [hunk 5]
```diff
@@ -73,6 +73,7 @@ def expDir = getInputParam('com.gradle.enterprise.build-validation.expDir')
 def expId = getInputParam('com.gradle.enterprise.build-validation.expId')
 def runId = getInputParam('com.gradle.enterprise.build-validation.runId')
 def runNum = getInputParam('com.gradle.enterprise.build-validation.runNum')
+def version = getInputParam('com.gradle.enterprise.build-validation.version')
```
Comment (etiennestuder) [inline components/scripts/gradle/gradle-init-scripts/configure-gradle-enterprise.gradle:76]: I would rename the variable to `scriptsVersion`.

File: components/capture-build-scan-url-maven-extension/src/main/java/com/gradle/CaptureBuildScansListener.java [hunk 6]
```diff
@@ -46,6 +46,9 @@ private static void addCustomDataOnBuildFinished(BuildScanApi buildScan) {
 
             String runId = System.getProperty("com.gradle.enterprise.build-validation.runId");
             addCustomValueAndSearchLink(buildScan, "Experiment run id", runId);
+
+            String version = System.getProperty("com.gradle.enterprise.build-validation.version");
```
Comment (etiennestuder) [inline components/capture-build-scan-url-maven-extension/src/main/java/com/gradle/CaptureBuildScansListener.java:50]: I would rename the variable to scriptsVersion.

**Review by etiennestuder — CHANGES_REQUESTED** [review]

**Review by etiennestuder — APPROVED** [review]

**Pull request merged**